### DISCUSSION TITLE
Agent output streams

### DISCUSSION THREAD

**Discussion by ChristianWeyer** [crewAIInc/crewAI#1116] [Feature Requests]
I just saw that AutoGen is planning to add an output streams feature beyond console output.
https://github.com/microsoft/autogen/pull/1290#discussion_r1460794605

This would be great if crewAI would provide a similar feature.

**Comment by joaomdmoura** [comment]
Interesting, curious to hear more about the use cases so we can get something that is great for those, anything comes to mind?

**Comment by ChristianWeyer** [comment]
One use case would be to stream, agent result to a web frontend.
Another could be to have really distributed agents working together.

**Comment by JaDenis** [comment]
I agree, i'd like to send answers via WebSockets to a web client.

**Comment by mdfederici** [comment]
I also want to be able to stream live output to a frontend.

Seems like it is also related to https://github.com/joaomdmoura/crewAI/issues/146

**Comment by bigr00** [comment]
Currently I found a workaround to capture terminal output while crew.kickoff() is working with subprocess and to stream the stdout. It’s ugly and gets unsupported characters, but at least can stream.

**Comment by JaDenis** [comment]
> Currently I found a workaround to capture terminal output while crew.kickoff() is working with subprocess and to stream the stdout. It’s ugly and gets unsupported characters, but at least can stream.

please, share the code
i'm currently doing it like this with "verbose=2" and it sucks
```python
from io import StringIO  # Python 3
import sys

# Create the in-memory "file"
temp_out = StringIO()

# Replace default stdout (terminal) with our stream
sys.stdout = temp_out

print("This is going in to the memory stream")
```

Sometimes i see agents questions to one another in console, i'd really like to capture these too
They write pretty high quality prompts into each other, gathering these would be 100% useful

**Comment by r-leyshon** [comment]
Capture stdout would be a dream. Capturing chain of thought reasoning for audit logs and such.

**Comment by smortezah** [comment]
Any news on this?

**Comment by khoi03** [comment]
If CrewAI could add streaming outputs, it would be a breakthrough

**Comment by smortezah** [comment]
I genuinely don’t want to use this:

```python
from typing import AsyncGenerator
import time

async def stream(text: str, chunk_size: int = 4, sleep: float = 0.03) -> AsyncGenerator:
    """Generator function to yield chunks of the final answer."""
    for i in range(0, len(text), chunk_size):
        time.sleep(sleep)
        yield text[i : i + chunk_size]

text = "This is a test message. This is a piece of text that will be streamed. This is the final message."
async for chunk in stream(text):
    print(chunk, end="")
```

**Comment by ChiefMateStarbuck** [comment]
+1 on this feature.

**Reply by NiuBlibing** [reply]
Anything update?

**Reply by ChiefMateStarbuck** [reply]
yeah, i switched to pydanticAI.

**Comment by JardelCunha** [comment]
+1
